rpc/src/test/scala/org/bitcoins/rpc/RpcClientTest.scala

```diff
@@ -0,0 +1,517 @@
+      assert(true)
+    }
+
+    client.getRawChangeAddress(Some("p2sh-segwit")).map { address =>
+      assert(true)
+    }
+
+    client.getRawChangeAddress(Some("bech32")).map { address =>
+      assert(true)
+    }
+  }
+
+  it should "be able to get the amount recieved by some address" in {
+    client.getNewAddress().flatMap { address =>
+      client.getReceivedByAddress(address).flatMap { amount =>
+        assert(amount == Bitcoins(0))
+      }
+    }
+  }
+
+  it should "be able to get the tx outset info" in {
+    client.getTxOutSetInfo.flatMap { info =>
+      client.getBlockCount.map { count =>
+        assert(info.height == count)
+      }
+      client.getBestBlockHash.map { hash =>
+        assert(info.bestblock == hash)
+      }
+    }
+  }
+
+  it should "be able to get the unconfirmed balance" in {
+    client.getUnconfirmedBalance.flatMap { balance =>
+      assert(balance == Bitcoins(0))
+      sendTransaction.flatMap { transaction =>
+        client.getUnconfirmedBalance.map { newBalance =>
+          assert(newBalance == transaction.amount)
+        }
+      }
+    }
+  }
+
+  it should "be able to refill the keypool" in {
+    client.getWalletInfo.flatMap { info =>
+      client.keyPoolRefill(info.keypoolsize + 1).flatMap { _ =>
+        client.getWalletInfo.flatMap { newInfo =>
+          assert(newInfo.keypoolsize == info.keypoolsize + 1)
+        }
+      }
+    }
+  }
+
+  it should "be able to get the block count" in {
+    val blockCountF = client.getBlockCount
+    blockCountF.map { count =>
+      logger.info(count.toString)
+      assert(count >= 0)
+    }
+  }
+
+  it should "be able to get the connection count" in {
+    val connectionCountF = client.getConnectionCount
+    connectionCountF.map { count =>
+      logger.info(count.toString)
+      assert(count == 0)
+    }
+  }
+
+  it should "be able to get the best block hash" in {
+    val bestHashF = client.getBestBlockHash
+    bestHashF.map { hash =>
+      logger.info(hash.toString)
+      assert(true)
+    }
+  }
+
+  it should "be able to get the mining info" in {
+    val miningInfoF = client.getMiningInfo
+    miningInfoF.map { info =>
+      logger.info(info.toString)
+      assert(info.chain == "regtest")
+    }
+  }
+
+  it should "be able to get the chain tips" in {
+    val chainTipsF = client.getChainTips
+    chainTipsF.map { tipArray =>
+      tipArray.foreach { tip =>
+        logger.info(tip.toString)
+        assert(tip.status == "active")
+      }
+      assert(true)
+    }
+  }
+
+  it should "be able to get the network info" in {
+    val networkInfoF = client.getNetworkInfo
+    networkInfoF.map { info =>
+      logger.info(info.toString)
+      assert(info.networkactive)
+      assert(info.connections == 0)
+      assert(info.localrelay)
+    }
+  }
+
+  it should "be able to generate blocks" in {
+    val blocksF = client.generate(3)
+    blocksF.map { blocks =>
+      blocks.foreach(block => logger.info(block.toString))
```

Comment:
log statement isn't needed. The best way to test this might be to call `.getBlockCount` and then generate 3 blocks and then call getblockcount again